six/modules/c++/samples/test_create_sidd_from_mem.cpp

```diff
@@ -1773,17 +1774,498 @@ void initProcessorInformation(
+            addExploitationFeatures(1);
+
+    siddBuilder.addProductProcessing();
+    siddBuilder.addDownstreamReprocessing();
+    siddBuilder.addErrorStatistics();
+    siddBuilder.addRadiometric();
+
+    //---------------------------------------------------------
+    // Take ownership of the SIDD data, the builder still can
+    // manipulate the same pointer after this happens if you
+    // want it to, but it won't try to release it when it goes
+    // out of scope if you steal() it.
+    //---------------------------------------------------------
+    return  std::auto_ptr<six::sidd::DerivedData>(siddBuilder.steal());
+}
+
+void initProductCreation(six::sidd::ProductCreation& productCreation)
+{
+    productCreation.productName = "ProductName";
+    productCreation.productClass = "Classy";
+    productCreation.classification.classification = "U";
+    productCreation.classification.compliesWith.push_back("ICD-710");
+    productCreation.classification.ownerProducer.push_back("ABW");
+    productCreation.classification.ownerProducer.push_back("AIA");
+    productCreation.classification.sciControls.push_back("HCS");
+    productCreation.classification.sciControls.push_back("SI");
+    productCreation.classification.sarIdentifier.push_back("EU");
+    productCreation.classification.sarIdentifier.push_back("BC");
+    productCreation.classification.disseminationControls.push_back("FOUO");
+    productCreation.classification.disseminationControls.push_back("IMC");
+    productCreation.classification.fgiSourceOpen.push_back("AIA");
+    productCreation.classification.fgiSourceOpen.push_back("ALB");
+    productCreation.classification.fgiSourceProtected.push_back("AIA");
+    productCreation.classification.fgiSourceProtected.push_back("AND");
+    productCreation.classification.releasableTo.push_back("ABW");
+    productCreation.classification.releasableTo.push_back("AFG");
+    productCreation.classification.nonICMarkings.push_back("SINFO");
+    productCreation.classification.nonICMarkings.push_back("DS");
+    productCreation.classification.classifiedBy = "PVT Snuffy";
+    productCreation.classification.compilationReason = "Testing purposes";
+    productCreation.classification.derivativelyClassifiedBy = "Anna J. Stepp";
+    productCreation.classification.classificationReason = "More testing";
+    productCreation.classification.nonUSControls.push_back("BOHEMIA");
+    productCreation.classification.nonUSControls.push_back("BALK");
+    productCreation.classification.derivedFrom = "Other documents";
+    productCreation.classification.declassDate.reset(new six::DateTime());
+    productCreation.classification.declassEvent = "N/A";
+    productCreation.classification.declassException = "25X1";
+
+    //SIDD 1.0 only
+    productCreation.classification.exemptedSourceType = "X8";
+    productCreation.classification.exemptedSourceDate.reset(new six::DateTime());
+
+    /*
+    This is SIDD_1.1 stuff.
```

Comment:
Keep all the SIDD 1.1 stuff in a branch for now... you might want to make a note of what all such changes are and then once this is merged into master, merge master into the SIDD 1.1 branch and put this stuff in.